PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs

```diff
@@ -157,29 +187,36 @@ public void AttachToRunspace(uint pid)
                 return;
             }
 
-            // Enter into to-attach process which will swap out the current runspace
+            // enter into to-attach process which will swap out the current runspace
             using (_currentPowerShell = PowerShell.Create())
             {
+                // scenario before entering, used to determine if we are local attaching
+                DebugScenario preScenario = GetDebugScenario();
+
+                _attachRequestEvent.Reset();
                 _currentPowerShell.Runspace = _runspace;
                 _currentPowerShell.AddCommand("Enter-PSHostProcess").AddParameter("Id", pid.ToString());
                 _currentPowerShell.Invoke();
 
-                // wait for invoke to finish swapping the runspaces
-                _attachRequestEvent.WaitOne(5000);
-
-                // make sure that the semaphore didn't just time out
-                if (GetDebugScenario() != DebugScenario.LocalAttach)
+                // wait for invoke to finish swapping the runspaces if you are attaching to a local process
+                if (preScenario != DebugScenario.RemoteSession)
                 {
-                    MessageBox.Show(Resources.ProcessAttachFailErrorBody, Resources.ProcessAttachFailErrorTitle, MessageBoxButton.OK, MessageBoxImage.Warning);
-                    ServiceCommon.Log("Failed to attach to running process.");
-                    return;
+                    _attachRequestEvent.WaitOne(DebugEngineConstants.AttachRequestEventTimeout);
+
+                    // scenario after entering, used to determine if semaphore timed out, should not match our pre-scenario

```

Comment:
I would prefer to use return value from waitOne() cause that would save people some time from reading comments and understand the implication behind the debugscenario check. 
I think I'm 100% sure the return value will be false for the case where the event was signaled before waitone is called, otherwise it would be a bug in that API...